File: samples/android/AdUnitsSample/src/main/java/com/facebook/samples/AdUnitsSample/fragments/InterstitialFragment.java
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 * All rights reserved.
 *
 * This source code is licensed under the license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.samples.AdUnitsSample.fragments;

import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;
import androidx.fragment.app.Fragment;
import com.facebook.ads.Ad;
import com.facebook.ads.AdError;
import com.facebook.ads.CacheFlag;
import com.facebook.ads.InterstitialAd;
import com.facebook.ads.InterstitialAdExtendedListener;
import com.facebook.ads.RewardData;
import com.facebook.common.preconditions.Preconditions;
import com.facebook.infer.annotation.Nullsafe;
import com.facebook.samples.AdUnitsSample.R;
import com.facebook.samples.ads.debugsettings.DebugToast;
import java.util.EnumSet;
import javax.annotation.Nullable;

@Nullsafe(Nullsafe.Mode.LOCAL)
public class InterstitialFragment extends Fragment implements InterstitialAdExtendedListener {

  private static final String TAG = InterstitialFragment.class.getSimpleName();

  private TextView interstitialAdStatusLabel;
  private Button loadInterstitialButton;
  private Button showInterstitialButton;
  @Nullable private InterstitialAd interstitialAd;

  private String statusLabel = "";

  @Override
  public View onCreateView(
      LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
    View view = inflater.inflate(R.layout.fragment_interstitial, container, false);

    // NULLSAFE_FIXME[Field Not Nullable]
    interstitialAdStatusLabel = (TextView) view.findViewById(R.id.interstitialAdStatusLabel);
    // NULLSAFE_FIXME[Field Not Nullable]
    loadInterstitialButton = (Button) view.findViewById(R.id.loadInterstitialButton);
    // NULLSAFE_FIXME[Field Not Nullable]
    showInterstitialButton = (Button) view.findViewById(R.id.showInterstitialButton);

    loadInterstitialButton.setOnClickListener(
        new View.OnClickListener() {
          @Override
          public void onClick(View view) {
            if (interstitialAd != null) {
              interstitialAd.destroy();
              interstitialAd = null;
            }
            setLabel("Loading interstitial ad...");

            // Create the interstitial unit with a placement ID (generate your own on the Facebook
            // app settings).
            // Use different ID for each ad placement in your app.
            interstitialAd =
                // NULLSAFE_FIXME[Parameter Not Nullable]
                new InterstitialAd(InterstitialFragment.this.getActivity(), "YOUR_PLACEMENT_ID");

            // Load a new interstitial.
            InterstitialAd.InterstitialLoadAdConfig loadAdConfig =
                interstitialAd
                    .buildLoadAdConfig()
                    // Set a listener to get notified on changes
                    // or when the user interact with the ad.
                    .withAdListener(InterstitialFragment.this)
                    .withCacheFlags(EnumSet.of(CacheFlag.VIDEO))
                    .withRewardData(new RewardData("YOUR_USER_ID", "YOUR_REWARD", 10))
                    .build();
            interstitialAd.loadAd(loadAdConfig);
          }
        });

    showInterstitialButton.setOnClickListener(
        new View.OnClickListener() {
          @Override
          public void onClick(View view) {
            if (interstitialAd == null
                || !interstitialAd.isAdLoaded()
                || interstitialAd.isAdInvalidated()) {
              // Ad not ready to show.
              setLabel("Ad not loaded. Click load to request an ad.");
            } else {
              // Ad was loaded, show it!
              setLabel("");
              interstitialAd.show();
            }
          }
        });
    return view;
  }

  @Override
  public void onDestroy() {
    if (interstitialAd != null) {
      interstitialAd.destroy();
      interstitialAd = null;
    }
    super.onDestroy();
  }

  @Override
  public void onError(Ad ad, AdError error) {
    if (ad == interstitialAd) {
      setLabel("Interstitial ad failed to load: " + error.getErrorMessage());
    }
  }

  @Override
  public void onAdLoaded(Ad ad) {
    if (ad == interstitialAd) {
      setLabel("Ad loaded. Click show to present!");
    }
  }

  @Override
  public void onInterstitialDisplayed(Ad ad) {
    showToast("Interstitial Displayed");
  }

  @Override
  public void onInterstitialDismissed(Ad ad) {
    showToast("Interstitial Dismissed");

    // Cleanup.
    Preconditions.checkNotNull(interstitialAd).destroy();
    interstitialAd = null;
  }

  @Override
  public void onAdClicked(Ad ad) {
    showToast("Interstitial Clicked");
  }

  @Override
  public void onLoggingImpression(Ad ad) {
    Log.d(TAG, "onLoggingImpression");
    showToast("Interstitial Impression");
  }

  private void setLabel(String label) {
    statusLabel = label;
    if (interstitialAdStatusLabel != null) {
      interstitialAdStatusLabel.setText(statusLabel);
    }
  }

  @Override
  public void onRewardedAdCompleted() {
    showToast("Reward Received");
  }

  @Override
  public void onRewardedAdServerSucceeded() {
    showToast("Server success!");
  }

  @Override
  public void onRewardedAdServerFailed() {
    showToast("Server failure");
  }

  @Override
  public void onInterstitialActivityDestroyed() {
    showToast("Activity destroyed");
  }

  private void showToast(String message) {
    Log.d(TAG, message);
    if (isAdded()) {
      DebugToast.show(requireActivity(), message, Toast.LENGTH_SHORT);
    }
  }
}

File: samples/android/KotlinAdUnitsSample/src/main/java/com/facebook/samples/adunitssamplekotlin/fragments/InterstitialFragment.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 * All rights reserved.
 *
 * This source code is licensed under the license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.samples.adunitssamplekotlin.fragments

import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView
import android.widget.Toast
import androidx.fragment.app.Fragment
import com.facebook.ads.*
import com.facebook.samples.adunitssamplekotlin.R
import java.util.*

class InterstitialFragment : Fragment(), InterstitialAdListener {

  companion object {
    val TAG = InterstitialFragment::class.java.simpleName
  }

  private var interstitialAd: InterstitialAd? = null

  private var interstitialAdStatusLabel: TextView? = null
  private var loadInterstitialButton: Button? = null
  private var showInterstitialButton: Button? = null

  override fun onCreateView(
      inflater: LayoutInflater,
      container: ViewGroup?,
      savedInstanceState: Bundle?
  ): View? {
    val view = inflater.inflate(R.layout.fragment_interstitial, container, false)
    interstitialAdStatusLabel = view?.findViewById(R.id.interstitialAdStatusLabel)
    loadInterstitialButton = view?.findViewById(R.id.loadInterstitialButton)
    showInterstitialButton = view?.findViewById(R.id.showInterstitialButton)

    loadInterstitialButton?.setOnClickListener {
      if (interstitialAd != null) {
        interstitialAd?.destroy()
        interstitialAd = null
      }
      setLabel("Loading interstitial ad...")

      interstitialAd = InterstitialAd(this.activity, "YOUR_PLACEMENT_ID")
      interstitialAd?.loadAd(
          interstitialAd!!
              .buildLoadAdConfig()
              .withAdListener(this)
              .withCacheFlags(EnumSet.of(CacheFlag.VIDEO))
              .build())
    }

    showInterstitialButton?.setOnClickListener {
      if (interstitialAd == null || !interstitialAd!!.isAdLoaded) {
        setLabel("Ad not loaded. Click load to request an ad.")
      } else {
        interstitialAd?.show()
        setLabel("")
      }
    }
    return view
  }

  override fun onDestroy() {
    interstitialAd?.destroy()
    interstitialAd = null

    super.onDestroy()
  }

  override fun onError(ad: Ad, error: AdError) {
    if (ad === interstitialAd) {
      setLabel("Interstitial ad failed to load: " + error.errorMessage)
    }
  }

  override fun onAdLoaded(ad: Ad) {
    if (ad === interstitialAd) {
      setLabel("Ad loaded. Click show to present!")
    }
  }

  override fun onInterstitialDisplayed(ad: Ad) {
    if (isAdded) {
      Toast.makeText(activity, "Interstitial Displayed", Toast.LENGTH_SHORT).show()
    }
  }

  override fun onInterstitialDismissed(ad: Ad) {
    if (isAdded) {
      Toast.makeText(activity, "Interstitial Dismissed", Toast.LENGTH_SHORT).show()
    }

    // Cleanup.
    interstitialAd?.destroy()
    interstitialAd = null
  }

  override fun onAdClicked(ad: Ad) {
    if (isAdded) {
      Toast.makeText(activity, "Interstitial Clicked", Toast.LENGTH_SHORT).show()
    }
  }

  override fun onLoggingImpression(ad: Ad) {
    Log.d(TAG, "onLoggingImpression")
  }

  private fun setLabel(label: String) {
    interstitialAdStatusLabel?.text = label
  }
}
